base_location_nuts/models/res_partner.py

```diff
@@ -1,61 +1,123 @@
-# -*- coding: utf-8 -*-
-##############################################################################
-# For copyright and license notices, see __openerp__.py file in root directory
-##############################################################################
-
-from openerp import models, fields, api
-from openerp.tools.translate import _
-import collections
 
+# -*- coding: utf-8 -*-
+# Copyright 2015 Antonio Espinosa <antonio.espinosa@tecnativa.com>
+# Copyright 2016 Jairo Llopis <jairo.llopis@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
 
-def dict_recursive_update(d, u):
-    for k, v in u.iteritems():
-        if isinstance(v, collections.Mapping):
-            r = dict_recursive_update(d.get(k, {}), v)
-            d[k] = r
-        else:
-            d[k] = u[k]
-    return d
+from odoo import api, fields, models
 
 
 class ResPartner(models.Model):
     _inherit = 'res.partner'
 
-    region = fields.Many2one(comodel_name='res.partner.nuts',
-                             string="Region")
-    substate = fields.Many2one(comodel_name='res.partner.nuts',
-                               string="Substate")
-    lbl_region = fields.Char(compute='_labels_get')
-    lbl_substate = fields.Char(compute='_labels_get')
+    nuts1_id = fields.Many2one(
+        comodel_name='res.partner.nuts', domain=[('level', '=', 1)],
+        string="NUTS L1")
+    nuts2_id = fields.Many2one(
+        comodel_name='res.partner.nuts', domain=[('level', '=', 2)],
+        string="NUTS L2", oldname="region")
+    nuts3_id = fields.Many2one(
+        comodel_name='res.partner.nuts', domain=[('level', '=', 3)],
+        string="NUTS L3", oldname="substate")
+    nuts4_id = fields.Many2one(
+        comodel_name='res.partner.nuts', domain=[('level', '=', 4)],
+        string="NUTS L4")
+
+    @api.multi
+    def _onchange_nuts(self, level):
+        field = self["nuts%d_id" % level]
```

Comment:
No, he is using the right thing. Attributes can get anything from the object, while keys can only get fields, so for security this is the recommended system to access fields with variable names, and even more when the variable comes from outside the method.

Please check slide 40 of https://www.odoo.com/es_ES/slides/slide/10-rules-for-safer-code-398